src/main/java/hudson/plugins/build_timeout/operations/AbortAndRestartOperation.java

```diff
@@ -0,0 +1,131 @@
+import hudson.plugins.build_timeout.BuildTimeOutOperation;
+import hudson.plugins.build_timeout.BuildTimeOutOperationDescriptor;
+
+import com.chikli.hudson.plugin.naginator.FixedDelay;
+
+/**
+ * Abort the build.
+ */
+public class AbortAndRestartOperation extends BuildTimeOutOperation {
+    
+    private final String maxRestarts;
+    
+    private static final Logger log = Logger.getLogger(AbortAndRestartOperation.class.getName());
+    
+    /**
+     * @return max restarts.
+     */
+    public String getMaxRestarts() {
+        return maxRestarts;
+    }
+        
+    @DataBoundConstructor
+    public AbortAndRestartOperation(String maxRestarts){
+        this.maxRestarts = maxRestarts;
+    }
+    
+    private static boolean isPresent() {
+        try {
+            Class.forName("com.chikli.hudson.plugin.naginator.NaginatorScheduleAction");
+            return true;
+        } catch (ClassNotFoundException ex) {
+            log.log(Level.FINEST, "Naginator not available. ", ex);
+            return false;
+        }
+    }
+    
+    private void rescheduleBuild(AbstractBuild<?, ?> build, BuildListener listener) {
+        FixedDelay sd = new FixedDelay(0); //Reschedule now!
+        
+        String maxRestartsStr = null;
+        try {
+            maxRestartsStr = build.getEnvironment(listener).expand(this.maxRestarts);
+        } catch (IOException e1) {
+            listener.error("Failed to expand environment variables.");
+            e1.printStackTrace(listener.getLogger());
+            return;
+        } catch (InterruptedException e1) {
+            listener.error("Failed to expand environment variables.");
+            e1.printStackTrace(listener.getLogger());
+            return;
+        }
+        
+        int maxRestarts = 0;
+        try {
+            maxRestarts = Integer.parseInt(maxRestartsStr);
+        } catch (NumberFormatException e) {
+            listener.error("Invalid Maximum restarts: {0}", maxRestartsStr);
+            e.printStackTrace(listener.getLogger());
+            return;
+        }
+        build.addAction(new com.chikli.hudson.plugin.naginator.NaginatorScheduleAction(maxRestarts, sd, false));
+        listener.getLogger().println(Messages.AbortAndRestartOperation_ScheduledRestart(maxRestarts));
+    }
+    
+    /**
+     * @param build
+     * @param listener
+     * @param effectiveTimeout
+     * @return
+     * @see hudson.plugins.build_timeout.BuildTimeOutOperation#perform(hudson.model.AbstractBuild, hudson.model.BuildListener, long)
+     */
+    @Override
+    public boolean perform(AbstractBuild<?, ?> build, BuildListener listener, long effectiveTimeout) {
+        if (isPresent()) {
+            rescheduleBuild(build, listener);
+        } else {
+            listener.error(Messages.AbortAndRestartOperation_InstallNaginator());
+        }
+        
+        return new AbortOperation().perform(build, listener, effectiveTimeout);
+    }
+   
+    @Extension 

```

Comment:
@ikedam This needs to be marked `@Extension(optional = true)` (or the dependency in the POM be made non-optional), otherwise https://gist.github.com/rtyler/06457d7d9d4d4eccef36f6d9aa1c9738